user-operator/src/test/java/io/strimzi/operator/user/operator/SimpleAclOperatorTest.java

```diff
@@ -62,162 +71,187 @@ public static void after() {
-        SimpleAclRule rule1 = new SimpleAclRule(AclRuleType.ALLOW, resource, "*", AclOperation.WRITE);
-
-        KafkaPrincipal foo = new KafkaPrincipal("User", "CN=foo");
-        Acl readAcl = new Acl(foo, Allow$.MODULE$, "*", Read$.MODULE$);
-        scala.collection.immutable.Set<Acl> readAclSet = new scala.collection.immutable.Set.Set1<>(readAcl);
-        Acl writeAcl = new Acl(foo, Allow$.MODULE$, "*", Write$.MODULE$);
-        scala.collection.immutable.Set<Acl> writeAclSet = new scala.collection.immutable.Set.Set1<>(writeAcl);
-
-        Resource resource1 = new Resource(Topic$.MODULE$, "my-topic", PatternType.LITERAL);
-        Resource resource2 = new Resource(Topic$.MODULE$, "my-topic2", PatternType.LITERAL);
+        ResourcePattern resource1 = new ResourcePattern(ResourceType.TOPIC, "my-topic", PatternType.LITERAL);
+        ResourcePattern resource2 = new ResourcePattern(ResourceType.TOPIC, "my-topic2", PatternType.LITERAL);
 
-        scala.collection.immutable.Map<Resource, scala.collection.immutable.Set<Acl>> map = new scala.collection.immutable.Map.Map1<>(resource1, readAclSet);
-        ArgumentCaptor<KafkaPrincipal> principalCaptor = ArgumentCaptor.forClass(KafkaPrincipal.class);
-        when(mockAuthorizer.getAcls(principalCaptor.capture())).thenReturn(map);
+        KafkaPrincipal foo = new KafkaPrincipal(KafkaPrincipal.USER_TYPE, "CN=foo");
+        AclBinding readAclBinding = new AclBinding(resource1, new AccessControlEntry(foo.toString(), "*", org.apache.kafka.common.acl.AclOperation.READ, AclPermissionType.ALLOW));
+        AclBinding writeAclBinding = new AclBinding(resource2, new AccessControlEntry(foo.toString(), "*", org.apache.kafka.common.acl.AclOperation.WRITE, AclPermissionType.ALLOW));
 
-        ArgumentCaptor<scala.collection.immutable.Set<Acl>> aclCaptor = ArgumentCaptor.forClass(scala.collection.immutable.Set.class);
-        ArgumentCaptor<Resource> resourceCaptor = ArgumentCaptor.forClass(Resource.class);
-        doNothing().when(mockAuthorizer).addAcls(aclCaptor.capture(), resourceCaptor.capture());
+        SimpleAclRuleResource resource = new SimpleAclRuleResource("my-topic2", SimpleAclRuleResourceType.TOPIC, AclResourcePatternType.LITERAL);
+        SimpleAclRule rule1 = new SimpleAclRule(AclRuleType.ALLOW, resource, "*", AclOperation.WRITE);
 
-        ArgumentCaptor<scala.collection.immutable.Set<Acl>> deleteAclCaptor = ArgumentCaptor.forClass(scala.collection.immutable.Set.class);
-        ArgumentCaptor<Resource> deleterResourceCaptor = ArgumentCaptor.forClass(Resource.class);
-        when(mockAuthorizer.removeAcls(deleteAclCaptor.capture(), deleterResourceCaptor.capture())).thenReturn(true);
+        ArgumentCaptor<Collection<AclBinding>> aclBindingsCaptor = ArgumentCaptor.forClass(Collection.class);
+        ArgumentCaptor<Collection<AclBindingFilter>> aclBindingFiltersCaptor = ArgumentCaptor.forClass(Collection.class);
+        assertDoesNotThrow(() -> {
```

Comment:
I do not really understand why do we have here (and on the other places) the `assertDoesNotThrow` when just creating the mocks. Could you explain it please? (just here, not necessarily in the code)

Reply:
The mock creation methods call a `future.get` which raise a couple of exceptions (i.e. `InterruptedException`). The first version had a try - catch block failing the test in the catch, while after @samuel-hawker suggestion I moved to this more convenient way.